scipy/stats/tests/test_bootstrap.py

```diff
@@ -470,3 +472,230 @@ def statistic_1d(*data):
     res1 = statistic(x, y, z, axis=axis)
     res2 = statistic2(x, y, z, axis=axis)
     assert_allclose(res1, res2)
+
+
+# --- Test Monte Carlo Hypothesis Test --- #
+
+class TestMonteCarloHypothesisTest:
+    atol = 1e-2  # for comparing p-value
+
+    def test_input_validation(self):
+        # test that the appropriate error messages are raised for invalid input
+
+        def stat(x):
+            return stats.skewnorm(x).statistic
+
+        message = "`axis` must be an integer."
+        with pytest.raises(ValueError, match=message):
+            monte_carlo_test([1, 2, 3], stats.norm.rvs, stat, axis=1.5)
+
+        message = "`vectorized` must be `True` or `False`."
+        with pytest.raises(ValueError, match=message):
+            monte_carlo_test([1, 2, 3], stats.norm.rvs, stat, vectorized=1.5)
+
+        message = "`rvs` must be callable."
+        with pytest.raises(TypeError, match=message):
+            monte_carlo_test([1, 2, 3], None, stat)
+
+        message = "`statistic` must be callable."
+        with pytest.raises(TypeError, match=message):
+            monte_carlo_test([1, 2, 3], stats.norm.rvs, None)
+
+        message = "`n_resamples` must be a positive integer."
+        with pytest.raises(ValueError, match=message):
+            monte_carlo_test([1, 2, 3], stats.norm.rvs, stat,
+                             n_resamples=-1000)
+
+        message = "`n_resamples` must be a positive integer."
+        with pytest.raises(ValueError, match=message):
+            monte_carlo_test([1, 2, 3], stats.norm.rvs, stat,
+                             n_resamples=1000.5)
+
+        message = "`batch` must be a positive integer or None."
+        with pytest.raises(ValueError, match=message):
+            monte_carlo_test([1, 2, 3], stats.norm.rvs, stat, batch=-1000)
+
+        message = "`batch` must be a positive integer or None."
+        with pytest.raises(ValueError, match=message):
+            monte_carlo_test([1, 2, 3], stats.norm.rvs, stat, batch=1000.5)
+
+        message = "`alternative` must be in..."
+        with pytest.raises(ValueError, match=message):
+            monte_carlo_test([1, 2, 3], stats.norm.rvs, stat,
+                             alternative='ekki')
+
+    def test_batch(self):
+        # make sure that the `batch` parameter is respected by checking the
+        # maximum batch size provided in calls to `statistic`
+        np.random.seed(0)
```

Comment:
I think Pamphile tried to get rid of the legacy approach using RandomState. consider replacing it by `rng = np.random.default_rng(...)` and `x = rng.random(10)`

Reply:
Yes, yes... : P It's sort of a game now where I try to slide this thing under his radar, so if you don't personally care, I won't tell.

Reply:
🙈